### PR TITLE
[15.0][IMP] queue: test performance improvement

### PR DESCRIPTION
**Pull request by josep-tecnativa**
CC @Tecnativa TT45458

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

File: queue_job/tests/test_delayable.py [hunk 1]
```diff
@@ -8,9 +8,20 @@
 
 
 class TestDelayable(unittest.TestCase):
-    def setUp(self):
-        super().setUp()
-        self.recordset = mock.MagicMock(name="recordset")
+    @classmethod
+    def setUpClass(cls):
```
Comment (simahawk) [inline queue_job/tests/test_delayable.py:12]: IMO this change is useless. This class does not inherit from an odoo class and does not trigger any odoo functionality

Comment (chienandalu) [reply queue_job/tests/test_delayable.py:12]: Well, it avoids redeclaring the case for every test. Indeed not much gain here anyway and the context thing would be useless here, that's true

Comment (simahawk) [reply queue_job/tests/test_delayable.py:12]: It's a mocked object... changes nothing in terms of perf.
Anyway, I'd keep the setUpClass but I'd drop the ctx stuff because it gives the false idea that is needed for whatever  reason.

Comment (chienandalu) [reply queue_job/tests/test_delayable.py:12]: I agree :+1: Please, @josep-tecnativa :)

Comment (pedrobaeza) [reply queue_job/tests/test_delayable.py:12]: Well, maybe in terms of "consistency" to not wondering if put it or not.

Comment (chienandalu) [reply queue_job/tests/test_delayable.py:12]: The thing here is that this class test inherits directly from `unittest` so it's quite useless to set that

Comment (pedrobaeza) [reply queue_job/tests/test_delayable.py:12]: Yes, useless, but consistent to always read the same code everywhere in OCA (I wonder also why this is not using Odoo tests), and with minimum overhead. And the day this is switched to Odoo tests, you already have it.

Comment (simahawk) [reply queue_job/tests/test_delayable.py:12]: I disagree because you should always keep the setup minimal.
However.... Now that I think of this twice... I fear that since the refactoring of the test suite was done (14, 15, 16) these tests are not auto-discovered anymore (just search for this test class in the build).
Hence, we should use `BaseCase` at least.

Comment (pedrobaeza) [reply queue_job/tests/test_delayable.py:12]: @josep-tecnativa switch to `BaseCase` or `TransactionCase`, please.
